### PR TITLE
[CI:DOCS] Fix docs/version-check always requesting updates

### PR DESCRIPTION
**Pull request by cevich**
***DEPENDS ON:*** https://github.com/containers/podman/pull/17621

As suggested by @edsantiago, the complex script and direct-link in the
docs are adding little value while increasing the CI maintenance burden.
Simply retire the script and strip the direct (versioned) links from the
docs.

---

Fixes (in cirrus-cron CI), this message always appearing & job failing:

```
+ docs/version-check
LATEST=v4.4.1
version-check: docs/tutorials/mac_win_client.md: needs updating.
version-check: Please run this script in an interactive shell, and I'll fix it.
version-check: docs/tutorials/podman-for-windows.md: needs updating.
version-check: Please run this script in an interactive shell, and I'll fix it.
```

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by cevich** [comment]
@edsantiago PTAL when you have time, this is not critical but it is causing the daily cirrus-cron job on `main` to "fail".  I'll open another PR to disable the version-check for now until we're able to work out a fix (this PR or otherwise).

**Comment by cevich** [comment]
Force-push: Rebased on https://github.com/containers/podman/pull/17621 + added revert.

**Comment by cevich** [comment]
force-push: Fixed missing DCO

**Comment by edsantiago** [comment]
I'm going to defer this. It's impossible to review today, because there's a release in progress: tag without build.
```
version-check: FATAL: Windows EXE missing: https://github.com/containers/podman/releases/download/v4.4.2/podman-4.4.2-setup.exe
```
And `grep -o` is not an option I've ever used, so I'd like to get super-familiar with it before approving this. That won't happen today.

I'm totally fine with just removing this step. I don't know if it really adds any value.

**Comment by cevich** [comment]
> I'm totally fine with just removing this step. I don't know if it really adds any value.

In fact, now that I've thought about it, this is likely the case.  Not too long ago I updated the `Makefile` to rename the Windows installer file w/o a version number.  So I think this script can just be removed, and the docs updated to point at the Downloads page (where the unversioned Windows binary link should be maintained).

**Comment by cevich** [comment]
Correction: These markdown pages are about the released Windows installer, not the CI artifact.  Still, I think Ed's right, including the version number and a direct link adds little value and a whole lot of (script) complications.  Let's just remove it and the generic "release page" link is very likely good enough.

File: docs/tutorials/mac_win_client.md [hunk 1]
```diff
@@ -16,8 +16,7 @@ The remote client uses a client-server model. You need Podman installed on a Lin
 
 ### Windows
 
-Installing the Windows Podman client begins by downloading the Podman Windows installer. The Windows installer is built with each Podman release and is downloadable from its [release description page](https://github.com/containers/podman/releases/latest).  The Windows installer file is named `podman-#.#.#-setup.exe`, where the `#` symbols represent the version number of Podman.
-As of 2023-02-21 the latest version is [v4.4.1](https://github.com/containers/podman/releases/download/v4.4.1/podman-4.4.1-setup.exe).
+Installing the Windows Podman client begins by downloading the Podman Windows installer. The Windows installer is built with each Podman release and is downloadable from its [release description page](https://github.com/containers/podman/releases/latest).
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/mac_win_client.md:19]: I'm not sure people would figure out what to look for.  I think some pointers are helpful.

```suggestion
Installing the Windows Podman client begins by downloading the Podman Windows installer. The Windows installer is built with each Podman release and is downloadable from its [release description page](https://github.com/containers/podman/releases/latest). The Windows installer file is named `podman-#.#.#-setup.exe`, where the `#` symbols represent the version number of Podman.
```

**Comment by cevich** [comment]
Force-push: Rebased.

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons

**Pull request merged**